functions/supplier/zeromq-supplier/src/main/java/org/springframework/cloud/fn/supplier/zeromq/ZeroMqSupplierProperties.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2016-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.fn.supplier.zeromq;
+
+import java.time.Duration;
+
+import javax.validation.constraints.NotEmpty;
+
+import org.hibernate.validator.constraints.Range;
+import org.zeromq.SocketType;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.validation.annotation.Validated;
+
+/**
+ *
+ * @author Daniel Frey
+ * @since 3.1.0
+ */
+@ConfigurationProperties("zeromq.supplier")
+@Validated
+public class ZeroMqSupplierProperties {
+
+	/**
+	 * The {@link SocketType} the connection should make.
+	 */
+	private SocketType socketType = SocketType.SUB;
+
+	/**
+	 * Connection URL for to the ZeroMQ TCP Server.
+	 */
+	private String connectUrl;
+
+	/**
+	 * Bind Port for connecting to the ZeroMQ TCP Server.
+	 */
+	private int bindPort;
+
+	/**
+	 * The delay to consume from the ZeroMQ TCP Server when no data received.
```

Comment:
Let's don't talk about TCP at all! In fact we can connect via any other supported protocol in ZeroMq!
(Plus "server" remark again...)